### PR TITLE
Updates for finalfinal2 infrastructure since 1.0.0

### PR DESCRIPTION
**Pull request by lk86**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Explain your changes here.

Explain how you tested your changes here.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

Closes #0000
Closes #0000

### CODE REVIEW

File: automation/bake/Dockerfile [hunk 1]
```diff
@@ -11,20 +11,20 @@ RUN curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key --keyri
...
-RUN curl https://raw.githubusercontent.com/MinaProtocol/mina/${COMMIT_HASH}/automation/terraform/testnets/${TESTNET_NAME}/genesis_ledger.json -o "${CONFIG_FILE}"
+#RUN curl https://raw.githubusercontent.com/MinaProtocol/mina/${COMMIT_HASH}/automation/terraform/testnets/${TESTNET_NAME}/genesis_ledger.json -o "${CONFIG_FILE}"
 
-RUN head ${CONFIG_FILE}
+#RUN head ${CONFIG_FILE}
 
 # Create the defautl config dir and an empty config
 RUN mkdir -p /root/.mina-config && echo "{}" > /root/.mina-config/daemon.json
 
-RUN mina daemon -config-file ${CONFIG_FILE} -generate-genesis-proof true; \
```
Comment (es92) [inline automation/bake/Dockerfile:24]: I think this would remove the ability to have configs that aren't pre-baked into the genesis_ledger? (for example, for a future testnet?) I would vote we keep that functionality so keep this part of the bake script

File: automation/terraform/testnets/finalfinal2/finalfinal3_seeds.txt [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/dns4/seed-1.finalfinal2.o1test.net/tcp/10000/p2p/12D3KooWGb3tna8uSs1dyTVX5qMRbxQzTXmyGJMLw2hFZePCzSTg
```
Comment (es92) [inline automation/terraform/testnets/finalfinal2/finalfinal3_seeds.txt:1]: what is finalfinal2_seeds vs finalfinal3_seeds? (what is finalfinal3?)

File: automation/terraform/testnets/mainnet/genesis_ledger.json [hunk 3]
```diff
@@ -0,0 +1,19264 @@
+{
+  "genesis": {
+    "genesis_state_timestamp": "2021-03-06T00:00:00Z"
```
Comment (es92) [inline automation/terraform/testnets/mainnet/genesis_ledger.json:3]: This should be set to 4pm PT 3/16

File: automation/terraform/testnets/mainnet/main.tf [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  seed_count            = var.seed_count
+
+  upload_blocks_to_gcloud         = true
+  restart_nodes                   = false
+  restart_nodes_every_mins        = "60"
+  make_reports                    = true
+  make_report_every_mins          = "30"
+  make_report_discord_webhook_url = local.make_report_discord_webhook_url
+  make_report_accounts            = local.make_report_accounts
+  seed_peers_url                  = "https://storage.googleapis.com/seed-lists/mainnet_seeds.txt?456"
```
Comment (es92) [inline automation/terraform/testnets/mainnet/main.tf:129]: This will have to change to the foundation-owned bucket - I'll get that setup today

File: automation/terraform/monitoring/o1-testnet-alerts.tf [hunk 5]
```diff
@@ -14,7 +14,7 @@ module "o1testnet_alerts" {
 
   rule_filter            = "{testnet=~\".+\"}" # any non-empty testnet name
   rule_timeframe         = "1h"
-  pagerduty_alert_filter = "devnet"
+  pagerduty_alert_filter = "devnet|finalfinal2"
```
Comment (es92) [inline automation/terraform/monitoring/o1-testnet-alerts.tf:17]: |mainnet?

**Review by es92 — CHANGES_REQUESTED** [review]

File: src/lib/staged_ledger/staged_ledger.ml [hunk 6]
```diff
@@ -1568,8 +1568,7 @@ module T = struct
         (*There's enough work. Check if they satisfy other constraints*)
         Resources.budget_sufficient resources
       then
-        if Resources.space_constraint_satisfied resources then (resources, log)
-        else if Resources.worked_more ~constraint_constants resources then
+        if Resources.worked_more ~constraint_constants resources then
```
Comment (imeckler) [inline src/lib/staged_ledger/staged_ledger.ml:1571]: Why is this included?

**Review by imeckler — CHANGES_REQUESTED** [review]

File: automation/bake/Dockerfile (hunk 1 above)
Comment (lk86) [reply automation/bake/Dockerfile:24]: only production builds require baking as it is, as long as you set generateGenesisProof=true in the terraform, testnets can use images right from CI and the vast majority of networks are run that way.

I made this change for finalfinal2 so that the baked images were as close as possible to the original images in CI that have more testing, and so that using a baked docker image tests the default config file + entrypoint. I'm fine with un-commenting this before merging it back into master but I think this is a safer and more conservative way to bake images for production deployments/public networks.

File: automation/terraform/testnets/finalfinal2/finalfinal3_seeds.txt (hunk 2 above)
Comment (lk86) [reply automation/terraform/testnets/finalfinal2/finalfinal3_seeds.txt:1]: finalfinal3_seeds is the same file, there was a bad peer string in the first version I uploaded to google for finalfinal2 and for a few hours GCP was caching it, so i made finalfinal3 to avoid the cache hit and get people moving quickly. It shouldn't happen again if we properly verify community peer ID's before uploading

File: src/lib/staged_ledger/staged_ledger.ml (hunk 6 above)
Comment (lk86) [reply src/lib/staged_ledger/staged_ledger.ml]: I wanted to test a build with https://github.com/MinaProtocol/mina/pull/8098 included and I was expecting it to land in master before this finalfinal2 branch landed, but it hasn't yet. I just reverted the merge commit from 8098 so there are no longer .ml changes on this branch.

File: automation/terraform/testnets/mainnet/genesis_ledger.json (hunk 3 above)
Comment (lk86) [reply automation/terraform/testnets/mainnet/genesis_ledger.json:3]: this is not the mainnet ledger, its the finalfinal2 ledger with a new genesis timestamp so we can deploy something and run a test before the mainnet ledger is ready.
